hermes/src/aggregate.rs

```diff
@@ -179,12 +232,41 @@ pub async fn store_update(state: &State, update: Update) -> Result<()> {
     // we can build the message states
     build_message_states(state, accumulator_messages, wormhole_merkle_state).await?;
 
-    state.update_tx.send(()).await?;
+    // Update the aggregate state
+    let mut aggregate_state = state.aggregate_state.write().await;
+
+    // Check if the update is new or out of order
+    match aggregate_state.latest_completed_slot {
+        Some(latest_completed_slot) => {
+            if slot > latest_completed_slot {
+                aggregate_state.latest_completed_slot.replace(slot);
+                state
+                    .api_update_tx
+                    .send(AggregationEvent::New { slot })
+                    .await?;
+            } else {
+                state
+                    .api_update_tx
+                    .send(AggregationEvent::OutOfOrder { slot })
+                    .await?;
+            }
+        }
+        None => {
+            aggregate_state.latest_completed_slot.replace(slot);
+            state
+                .api_update_tx
+                .send(AggregationEvent::New { slot })
+                .await?;
+        }
+    }
```

Comment:
```suggestion
match aggregate_state.latest_completed_slot {
    Some(latest) if slot > latest => {
        aggregate_state.latest_completed_slot.replace(slot);
        state.api_update_tx.send(AggregationEvent::New { slot }).await?;
    }
    None => {
        aggregate_state.latest_completed_slot.replace(slot);
        state.api_update_tx.send(AggregationEvent::New { slot }).await?;
    }
    _ => {
        state.api_update_tx.send(AggregationEvent::OutOfOrder { slot }).await?;
    }
}
```

Really just a matter of taste but I think this is clearer.